### ISSUE TITLE
Gauge tick marks cannot be modified

### ISSUE THREAD

**Issue by countfloyd** [JFXtras/jfxtras-labs#11]
OK, let's try this one :)

I can't seem to modify tick marks at all. There are a lot of set methods to modify tick marks but they don't seem to do anything. For example, try setting the the max value to 1200 (gauge.setMaxValue(1200)). The gauge shows with too many ticks and labels, all showing on top of each other. If I try to limit how many ticks there are, it has no effect. I've tried the following:

```
        gauge.setMinorTicksVisible(false);
        gauge.setMajorTicksVisible(false);
        gauge.setMaxNoOfMajorTicks(10);
        gauge.setMajorTickSpacing(500);
```

No change.

**Comment by HanSolo** [comment]
Usually the niceScaling option is enabled by default which should give you a good looking scale. If this is switched to false you could adjust the scaling manually by using setMaxNoOfMinorTicks for example. And it seems that there is a problem with the manual scaling. Will take a closer look...

**Comment by HanSolo** [comment]
Hi there,

Hmm for me it works again...see the attached screenshot...
Here is the code for the left gauge...

```
    Radial gauge = new Radial();

    gauge.setNiceScaling(false);
    gauge.setTickLabelOrientation(Gauge.TicklabelOrientation.HORIZONTAL);
    gauge.setMajorTickSpacing(100);
    gauge.setMinorTicksVisible(true);
    gauge.setMajorTicksVisible(true);
    gauge.setMaxNoOfMinorTicks(100);
    gauge.setMaxValue(1200);


    Section[] areas = {
        new Section(gauge.getMaxValue() * 0.8, gauge.getMaxValue() * 0.9, Color.YELLOW),
        new Section(gauge.getMaxValue() * 0.9, gauge.getMaxValue(), Color.RED)
    };

    Section[] sections = {
        new Section(gauge.getMaxValue() * 0.2, gauge.getMaxValue() * 0.3, Color.BLUE),
        new Section(gauge.getMaxValue() * 0.3, gauge.getMaxValue() * 0.4, Color.LIME)
    };

    gauge.getGaugeModel().setAreas(areas);
    gauge.setAreasVisible(true);

    gauge.getGaugeModel().setSections(sections);
    gauge.setSectionsVisible(true);
    gauge.setTitle("no nice scale");
```

If the numbers in the ticklabels are getting too long you could also switch to another number format (see the other screenshot).

Could you maybe send me your complete code so that we could make sure we are using the same stuff...?

Cheers,

Gerrit

Am 12.07.2012 um 21:01 schrieb countfloyd:

**Comment by countfloyd** [comment]
Your example works for me in a simple program just showing the one gauge. I'm wondering if the problem is that I'm trying to modify this stuff after the gauge has already been added to the scene and displayed?

Try the following program and you'll see what I'm seeing (hopefully):

package test;

import javafx.application.Application;
import javafx.event.ActionEvent;
import javafx.event.EventHandler;
import javafx.scene.Scene;
import javafx.scene.SceneBuilder;
import javafx.scene.control.Button;
import javafx.scene.control.ButtonBuilder;
import javafx.scene.layout.BorderPane;
import javafx.scene.layout.BorderPaneBuilder;
import javafx.scene.paint.Color;
import javafx.stage.Stage;
import jfxtras.labs.scene.control.gauge.Gauge;
import jfxtras.labs.scene.control.gauge.Radial;
import jfxtras.labs.scene.control.gauge.Section;

public class TestMain extends Application {
    public void start(final Stage stage) throws Exception {
        final Radial gauge = new Radial();

```
    gauge.setNiceScaling(false);
    gauge.setTickLabelOrientation(Gauge.TicklabelOrientation.HORIZONTAL);
    gauge.setTitle("no nice scale");

    Button button = ButtonBuilder.create()
            .text("Set Gauge Parameters")
            .build();
    button.addEventHandler(ActionEvent.ACTION, new EventHandler<ActionEvent>() {
        public void handle(final ActionEvent event) {
            gauge.setMaxValue(1200);
            gauge.setMajorTickSpacing(100);
            gauge.setMinorTicksVisible(true);
            gauge.setMajorTicksVisible(true);
            gauge.setMaxNoOfMinorTicks(10);

            Section[] areas = {
                    new Section(gauge.getMaxValue() * 0.8, gauge.getMaxValue() * 0.9, Color.YELLOW),
                    new Section(gauge.getMaxValue() * 0.9, gauge.getMaxValue(), Color.RED)
            };

            Section[] sections = {
                    new Section(gauge.getMaxValue() * 0.2, gauge.getMaxValue() * 0.3, Color.BLUE),
                    new Section(gauge.getMaxValue() * 0.3, gauge.getMaxValue() * 0.4, Color.LIME)
            };

            gauge.getGaugeModel().setAreas(areas);
            gauge.setAreasVisible(true);

            gauge.getGaugeModel().setSections(sections);
            gauge.setSectionsVisible(true);

        }
    });

    BorderPane root = BorderPaneBuilder.create()
            .center(gauge)
            .top(button)
            .build();

    Scene scene = SceneBuilder.create()
            .width(1024)
            .height(768)
            .root(root)
            .build();
    stage.setResizable(true);
    stage.setTitle("Gauge");
    stage.setScene(scene);
    stage.centerOnScreen();
    stage.show();
}

public static void main(String[] args) {
    launch(args);
}
```

}

**Comment by HanSolo** [comment]
Ok, now I see the problem too, it seems that there are some event listeners missing...will fix it...and keep you updated...

Cheers and thanx for the help,

Gerrit

Am 12.07.2012 um 23:59 schrieb countfloyd:

**Comment by HanSolo** [comment]
Problem seems to be fixed (in master-2.2 branch). Here is the code (based on yours just with a little modification) that I've used to test it...

import javafx.application.Application;
import javafx.event.ActionEvent;
import javafx.event.EventHandler;
import javafx.scene.Scene;
import javafx.scene.SceneBuilder;
import javafx.scene.control.Button;
import javafx.scene.control.ButtonBuilder;
import javafx.scene.layout.BorderPane;
import javafx.scene.layout.BorderPaneBuilder;
import javafx.scene.paint.Color;
import javafx.stage.Stage;
import jfxtras.labs.scene.control.gauge.Gauge;
import jfxtras.labs.scene.control.gauge.Radial;
import jfxtras.labs.scene.control.gauge.Section;

/**
- Created by
- User: hansolo
- Date: 13.07.12
- Time: 08:12
  */
  public class TestMain extends Application {
  public void start(final Stage stage) throws Exception {
      final Radial gauge = new Radial();
  
  ```
  gauge.setNiceScaling(false);
  gauge.setTickLabelOrientation(Gauge.TicklabelOrientation.HORIZONTAL);
  gauge.setTitle("no nice scale");
  gauge.setPrefSize(400, 400);
  
  Button button = ButtonBuilder.create()
          .text("Set Gauge Parameters")
          .build();
  button.addEventHandler(ActionEvent.ACTION, new EventHandler<ActionEvent>() {
      public void handle(final ActionEvent event) {
          gauge.setMaxValue(1200);
          gauge.setMajorTickSpacing(100);       // used if niceScaling == false
          gauge.setMinorTicksVisible(true);
          gauge.setMajorTicksVisible(true);
                          // gauge.setMaxNoOfMinorTicks(100) // used if niceScaling == true
          gauge.setMinorTickSpacing(10);        // used if niceScaling == false
  
          Section[] areas = {
                  new Section(gauge.getMaxValue() * 0.8, gauge.getMaxValue() * 0.9, Color.YELLOW),
                  new Section(gauge.getMaxValue() * 0.9, gauge.getMaxValue(), Color.RED)
          };
          gauge.getGaugeModel().setAreas(areas);
          gauge.setAreasVisible(true);
  
          Section[] sections = {
                  new Section(gauge.getMaxValue() * 0.2, gauge.getMaxValue() * 0.3, Color.BLUE),
                  new Section(gauge.getMaxValue() * 0.3, gauge.getMaxValue() * 0.4, Color.LIME)
          };
  
  
          gauge.getGaugeModel().setSections(sections);
          gauge.setSectionsVisible(true);
      }
  });
  
  BorderPane root = BorderPaneBuilder.create()
                                     .center(gauge)
                                     .top(button)
                                     .build();
  
  Scene scene = SceneBuilder.create()
                            .width(1024)
                            .height(768)
                            .root(root)
                            .build();
  
  stage.setResizable(true);
  stage.setTitle("Gauge");
  stage.setScene(scene);
  stage.centerOnScreen();
  stage.show();
  ```
  
  }
  
  public static void main(String[] args) {
      launch(args);
  }
  }
